### PR TITLE
Add log gathering to debug command.

### PR DESCRIPTION
**Pull request by ncabatoff**

### CODE REVIEW

**Review by swayne275 — APPROVED** [review]

File: command/debug.go [hunk 1]
```diff
@@ -981,3 +991,29 @@ func (c *DebugCommand) captureError(target string, err error) {
 	})
 	c.errLock.Unlock()
 }
+
+func (c *DebugCommand) writeLogs(ctx context.Context) error {
+	out, err := os.Create(filepath.Join(c.flagOutput, "vault.log"))
```
Comment (raskchanky) [inline command/debug.go:996]: Do we need to close `out`?

**Review by briankassouf — APPROVED** [review]

File: command/debug.go [hunk 2]
```diff
@@ -981,3 +991,29 @@ func (c *DebugCommand) captureError(target string, err error) {
...
+				return nil
+			}
+			_, err = out.WriteString(log)
+			if err != nil {
+				return err
+			}
+		case <-ctx.Done():
+			return nil
+		}
+	}
```
Comment (raskchanky) [inline command/debug.go:1018]: What causes this loop to terminate other than errors? I don't think the context will time out, will it?

**Review by raskchanky — APPROVED** [review]
Looks good, just a few questions

File: command/debug.go (hunk 1 above)
Comment (ncabatoff) [reply command/debug.go:996]: D'oh, ty!

File: command/debug.go (hunk 2 above)
Comment (ncabatoff) [reply command/debug.go:1018]: Yes, the context passed in has a timeout.

**Pull request merged**